src/clean-core/format.hh

```diff
@@ -0,0 +1,167 @@
+#pragma once
+
+#include <type_traits>
+
+#include <clean-core/function_ptr.hh>
+#include <clean-core/span.hh>
+#include <clean-core/string_stream.hh>
+#include <clean-core/string_view.hh>
+#include <clean-core/typedefs.hh>
+
+namespace cc
+{
+template <class T>
+struct arg;
```

Comment:
why is this in the "global" `cc` namespace?

Reply:
This is how you define named arguments. ` cc::format("{h} {w}!", cc::arg("w", "World"), cc::arg("h", "Hello"));`

Reply:
`cc::arg` was renamed to `cc::format_arg` and a user defined literal was added for better ease of use.